### PR TITLE
Darktrace Autoscaling vSensor Template

### PR DESCRIPTION
**Pull request by mstratford-dt**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

* Adds new Darktrace Autoscaling vSensor template

### CODE REVIEW

**Comment by mstratford-dt** [comment]
@microsoft-github-policy-service agree [company="Darktrace Holdings Ltd"]

**Comment by mstratford-dt** [comment]
Hi @alex-frankel, i've not found anywhere a documented process for who to assign to get this reviewed for us. Would you be able to have a look or reassign? Many thanks!

File: application-workloads/darktrace/darktrace-vsensor-autoscaling/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,1141 @@
...
...g/lock /var/lib/apt/lists/lock /var/cache/apt/archives/lock >/dev/null 2>&1 && [ $tries -le $maxtries ]; do\n       sleep 20\n       ((tries=tries+1))\n    done\n    if [ $tries -ge $maxtries ]; then\n        echo \"ERROR: Failed to get apt / dpkg lock before timeout. Please wait and try again, or fix dpkg lock manually.\"\n        exit 1\n    fi\n}\nexec > >(tee -a /var/log/user-data.log|logger -t user-data -s 2>/dev/console) 2>&1\nif [ \"', parameters('VSensorUpdateKey'), '\" = \"updateKey-example\" ]; then\n    echo \"Skipping install in azure CI due to dummy key.\"\n    exit 0\nfi\nwait_for_apt\nsleep 10\nwait_for_apt\nbash <(wget https://packages.darktrace.com/install -O -) --updateKey \"', parameters('VSensorUpdateKey'), '\"\n/usr/sbin/set_pushtoken.sh \"', parameters('appliancePushtoken'), '\" \"', parameters('applianceHostName'), ':', parameters('appliancePort'), '\"\nset_ossensor_hmac.sh \"', parameters('OSSensorHMACToken'), '\"\nset_pcap_azure_container.sh \"', variables('blobStorageAccountName'), '\" \"', variables('blobStorageContainerName'), '\"\nset_ossensor_loadbalancer_direct.sh 1\nset_ephemeral.sh 1\necho \"Configuration complete, vSensor is ready for use.\"\n'))]"
```
Comment (alex-frankel) [inline application-workloads/darktrace/darktrace-vsensor-autoscaling/azuredeploy.json:651]: are these bash scripts getting pulled from `packages.darktrace.com`?

Comment (mstratford-dt) [reply application-workloads/darktrace/darktrace-vsensor-autoscaling/azuredeploy.json:651]: Yes! Thanks for reviewing.

**Pull request merged**